### PR TITLE
Add api for recent activity in graphql

### PR DESCRIPTION
**Pull request by Rup-Narayan-Rajbanshi**
## Changes

- Add api for recent activity in graphql
Mention related users here if any.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [x] permission checks (tests here too)
- [x] translations

### CODE REVIEW

File: apps/project/schema.py [hunk 1]
```diff
@@ -116,6 +116,21 @@ def get_top_entity_contributor(project, Entity):
     ]
 
 
+class RecentActivityType(graphene.ObjectType):
+    id = graphene.Int()
+    created_at = graphene.DateTime()
+    project = graphene.Int()
```
Comment (thenav56) [inline apps/project/schema.py]:
```suggestion
    project = graphene.ID(required=True)
```

File: apps/project/schema.py [hunk 2]
```diff
@@ -116,6 +116,21 @@ def get_top_entity_contributor(project, Entity):
     ]
 
 
+class RecentActivityType(graphene.ObjectType):
+    id = graphene.Int()
```
Comment (thenav56) [inline apps/project/schema.py]:
```suggestion
    id = graphene.ID(required=True)
```

File: apps/project/schema.py [hunk 3]
```diff
@@ -116,6 +116,21 @@ def get_top_entity_contributor(project, Entity):
...
+    created_at = graphene.DateTime()
+    project = graphene.Int()
+    project_display_name = graphene.String()
+    created_by = graphene.Int()
```
Comment (thenav56) [inline apps/project/schema.py]:
```suggestion
    created_by = graphene.ID()
```

File: apps/project/schema.py [hunk 4]
```diff
@@ -116,6 +116,21 @@ def get_top_entity_contributor(project, Entity):
...
+    created_by_display_picture = graphene.String()
+    type = graphene.String()
```
Comment (thenav56) [inline apps/project/schema.py]: Let's use enum instead of String.

File: apps/project/schema.py (hunk 3 above)
Comment (thenav56) [inline apps/project/schema.py]: Maybe we can just use usertype here.

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/project/models.py [hunk 5]
```diff
@@ -27,6 +27,12 @@
 from datetime import timedelta
 
 
+class RecentActivityType(models.TextChoices):
+    LEAD = 'lead', 'lead'
+    ENTRY = 'entry', 'entry'
+    ENTRY_COMMENT = 'entry-comment', 'entry-comment'
```
Comment (thenav56) [inline apps/project/models.py]:
```suggestion
    LEAD = 'lead', 'Source'
    ENTRY = 'entry', 'Entry'
    ENTRY_COMMENT = 'entry-comment', 'Entry Comment'
```

File: apps/project/schema.py [hunk 6]
```diff
@@ -295,6 +297,26 @@ def resolve_regions(root, info, **kwargs):
...
+    def resolve_created_by(root, info, **kwargs):
+        id = int(root['created_by'])
+        return info.context.dl.project.users.load(id)
+
+    def resolve_project(root, info, **kwargs):
+        id = int(root['project'])
+        return info.context.dl.project.projects.load(id)
+
+    def resolve_type_display(root, info, **kwargs):
+        return root['type']
```
Comment (thenav56) [inline apps/project/schema.py]: Can you confirm this provides label instead of value?

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/project/schema.py (hunk 6 above)
Comment (Rup-Narayan-Rajbanshi) [reply apps/project/schema.py]: @thenav56 , here type is not generated from Choice fieds insead it is generate as models.Value('entry-comment', output_field=models.CharField()), so it return text string.

**Review by thenav56 — APPROVED** [review]

**Comment by AdityaKhatri** [comment]
Closed because it is included in this: https://github.com/the-deep/server/pull/1128/commits

**Pull request closed without merging**